slatedb/src/garbage_collector/compacted_gc.rs

```diff
@@ -100,28 +99,38 @@ impl CompactedGcTask {
         Ok(max_l0_ts)
     }
 
-    /// Returns a `DateTime<Utc>` barrier based on the compactor's oldest running compaction start.
+    /// Returns the oldest recent compaction start time from persisted state. "Recent" currently
+    /// means all running compactions and the most recently finished compaction.
```

Comment:
this is a bit of a mouthful 😅 I think we should be a little more explicit: "Returns the minimum _starting_ timestamp of (a) all of the ongoing compactions and (b) the most recently completed compaction". 

Would be helpful to document why this is the algorithm inline (i.e. why is the most recently completed compaction relevant?) Something like "Any SST with a timestamp less than this value is the result of a complete compaction and therefore eligible for garbage collection"

Reply:
fbfac956